chore: add Slack notification when merge-train is dequeued

johnathan79717 opened the pull request:
## Summary
- Adds a GitHub Actions workflow that notifies #honk-team when the `merge-train/barretenberg` PR is removed from the merge queue

## Motivation
This helps get people to look at it sooner when it's dequeued.

## Test plan
- Merge to next, then verify notification fires when the merge-train/barretenberg PR is dequeued

ludamad commented:
Thanks. The ideal is if it's shared the script that already slack notifies people. We should be able to call it through the runner with a special flag here. Basically it would be good if this worked for every merge train and used the data there for which channel to notify.

johnathan79717 commented:
> Thanks. The ideal is if it's shared the script that already slack notifies people. We should be able to call it through the runner with a special flag here. Basically it would be good if this worked for every merge train and used the data there for which channel to notify.

Done. Thanks for the suggestion.

ludamad commented:
Pushed a simplification for this being called outside of ci3 :)

johnathan79717 committed:
chore: add Slack notification when merge-train is dequeued

## Summary
- Adds a GitHub Actions workflow that notifies #honk-team when the `merge-train/barretenberg` PR is removed from the merge queue

## Motivation
This helps get people to look at it sooner when it's dequeued.

## Test plan
- Merge to next, then verify notification fires when the merge-train/barretenberg PR is dequeued
